cilium-cli/connectivity/tests/errors.go

```diff
@@ -175,17 +170,23 @@ func (n *noErrorsInLogs) Run(ctx context.Context, t *check.Test) {
 				if err != nil {
 					a.Fatalf("Error reading Cilium logs: %s", err)
 				}
-				n.checkErrorsInLogs(id, logs.Bytes(), a, &opts)
+				_ = n.checkErrorsInLogs(id, logs.Bytes(), a, &opts)
 
 				if restarts > 0 && !ignore {
-					a.Failf("Non-zero (%d) restart count of %s must be investigated", restarts, id)
-
-					logs = bytes.Buffer{}
-					err := client.GetLogs(ctx, pod.Namespace, pod.Name, container, prevOpts, &logs)
-					if err == nil {
-						n.checkErrorsInLogs(id, logs.Bytes(), a, &prevOpts)
+					shouldFail := true
+					// Fetch previous logs once here and analyze before deciding to fail.
+					var prevLogs bytes.Buffer
+					if err := client.GetLogs(ctx, pod.Namespace, pod.Name, container, prevOpts, &prevLogs); err != nil {
+						a.Logf("Warning: Could not retrieve previous logs to check for legitimate operator restart patterns: %s", err)
```

Comment:
This log message talks about operator but the statement is run for all Pods where there is a nonzero restart count. The error message should not make such specific references if it doesn't make sense for the other cases.

For consistency we can probably just follow the same pattern as other similar failures and use `Failf()`.
```suggestion
						a.Failf("Error reading previous Cilium logs: %s", err)
```

Reply:
ah, that makes sense! Thanks